WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -829,6 +812,25 @@ final class EditableOrderViewModel: ObservableObject {
     func onAddCustomAmountButtonTapped() {
         analytics.track(.orderCreationAddCustomAmountTapped)
     }
+
+    func addCustomAmountViewModel(with option: OrderCustomAmountsSection.ConfirmationOption?) -> AddCustomAmountViewModel {
+        let viewModel = AddCustomAmountViewModel(inputType: addCustomAmountInputType(from: option ?? .fixedAmount),
+                                                 onCustomAmountEntered: { [weak self] amount, name, feeID, isTaxable in
+            let taxStatus: OrderFeeTaxStatus = isTaxable ? .taxable : .none
+            if let feeID = feeID {
+                self?.updateFee(with: feeID, total: amount, name: name, taxStatus: taxStatus)
+            } else {
+                self?.addFee(with: amount, name: name, taxStatus: taxStatus)
+            }
+        })
+
+        if let editingFee {
+            viewModel.preset(with: editingFee)
+            self.editingFee = nil
+        }
+
+        return viewModel
+    }
```

Comment:
Yeah, these need to be in the `EditableOrderViewModel` because this provides all information related to the order to the custom amount view model (i.e. editing fee information, base amount for percentage), and reacts to the callbacks by adding fees to the order. We follow the same approach with all the other elements, coupons, notes, customer data, etc. 
By the way, we were already creating the `AddCustomAmountsViewModel` in the `EditableOrderViewModel`, only that now we moved it to a function instead of being a computed property.